### PR TITLE
Bugfix 227

### PR DESCRIPTION
**Pull request by NicholasDanks**
@soumyaray to review

### CODE REVIEW

File: R/evaluate_effects.R [hunk 1]
```diff
@@ -78,21 +78,23 @@ fSquared <- function(seminr_model, iv, dv) {
...
-      | any(names(seminr_model$measurement_model) == "scaled_interaction" )) {
-    return("The fSquare cannot be calculated as the model contains an interaction term and omitting either the antecedent or moderator in the interaction term will cause model estimation to fail")
-  }
-
   path_matrix <- seminr_model$path_coef
   fsquared_matrix <- path_matrix
-
   for (dv in all_endogenous(seminr_model$smMatrix)) {
-    for (iv in all_exogenous(seminr_model$smMatrix)) {
+    ifelse(length(grep("\\*",seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1]) ) > 0,
```
Comment (soumyaray) [inline R/evaluate_effects.R:84]: What does `seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1]` do?  Could we factor it out as a new function for inspecting structural matrix in the file R/inspect_smMatrix.R ?

Comment (soumyaray) [reply R/evaluate_effects.R:84]: `length(grep("\\*",seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1]) ) > 0` could also be just called `exists_interactions(seminr_model$smMatrix)`

Comment (soumyaray) [reply R/evaluate_effects.R:84]: And overall, see if we can write such code without manually digging through smMatrix

File: R/evaluate_effects.R [hunk 2]
```diff
@@ -78,21 +78,23 @@ fSquared <- function(seminr_model, iv, dv) {
...
+      int_components <- unique(unlist(strsplit(seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1][grep("\\*",seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1])], "\\*"))),
+      int_components <- NA)
+    for (iv in setdiff(all_exogenous(seminr_model$smMatrix), int_components)) {
       fsquared_matrix[iv, dv] <- fSquared(seminr_model = seminr_model,
                                           iv = iv,
                                           dv = dv)
+      fsquared_matrix[int_components,dv] <- NA
     }
   }
+  if (any(names(seminr_model$measurement_model) == "orthogonal_interaction")
```
Comment (soumyaray) [inline R/evaluate_effects.R:94]: isn't this if condition also just checking if interactions exist? why two ways of doing it (looking through smMatrix, and looking and measurement_model)?

File: R/evaluate_effects.R [hunk 3]
```diff
@@ -97,4 +97,5 @@ model_fsquares <- function(seminr_model) {
     comment(fsquared_matrix) <- "The fSquare for certain relationships cannot be calculated as the model contains an interaction term and omitting either the antecedent or moderator in the interaction term will cause model estimation to fail"
   }
   convert_to_table_output(fsquared_matrix)
+
```
Comment (soumyaray) [inline R/evaluate_effects.R:100]: blank line?

**Review by soumyaray — COMMENTED** [review]
Just some suggestions to try using (or creating) inspect methods in R/inspect_smMatrix.R and R/inspect_mmMatrix.R instead of writing complex code that digs through them.

File: R/evaluate_effects.R (hunk 1 above)
Comment (NicholasDanks) [reply R/evaluate_effects.R:84]: ```seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1]``` is replaced by antecedents_of()

Comment (NicholasDanks) [reply R/evaluate_effects.R:84]: ```grep("\\*",seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1])```  ```becomes interactions_of(dv, seminr_model$smMatrix)```

Comment (soumyaray) [reply R/evaluate_effects.R:84]: quick suggestion: how about `all_interactions`? `interactions_of` sounds like we are going to specify a construct whose interactions we want?

Comment (NicholasDanks) [reply R/evaluate_effects.R:84]: Yup, that's exactly what we are doing in this function. all_interactions is next :)

Comment (soumyaray) [reply R/evaluate_effects.R:84]: brilliant! shoulda looked closer!

**Pull request merged**